apps/src/lib/kits/maker/boards/microBit/LightSensor.js

```diff
@@ -86,13 +86,11 @@ export default class LightSensor extends EventEmitter {
 
   // Reset the state to initial values between runs
   reset() {
-    this.state = {
-      threshold: 128,
-      rangeMin: 0,
-      rangeMax: MAX_LIGHT_SENSOR_VALUE,
-      currentReading: 0,
-      currentBufferWriteIndex: 0
-    };
+    this.state.threshold = 128;
+    this.state.rangeMin = 0;
+    this.state.rangeMax = MAX_LIGHT_SENSOR_VALUE;
+    this.state.currentReading = 0;
+    this.state.currentBufferWriteIndex = 0;
```

Comment:
It might be handy to use spread syntax here for two reasons: 
1. it will make it more clear that we want things in the existing state to be carried into the new state; 
2. by creating a new state object that uses some of the existing one, we embrace immutability, which has many advantages, and you might recognise from state mutations in React.